lib/Agrammon/OutputFormatter/PDF.pm6

```diff
@@ -36,6 +36,8 @@ sub create-pdf($temp-dir-name, $pdf-prog, $timeout, $username, $dataset-name, %d
     }
 
     my $filename = "agrammon_export_" ~ $username ~ "_$dataset-name";
+    # cannot have / in filenames
+    $filename ~~ s:g/ \/ / - /;
```

Comment:
Wonder if it's worth having an allow list instead (e.g. `s:g/<-[\w\s_-]>/-/` or so), given exactly what different OSes permit is quite variable...

Reply:
> Wonder if it's worth having an allow list instead (e.g. `s:g/<-[\w\s_-]>/-/` or so), given exactly what different OSes permit is quite variable...

I figured this was good enough, as the server is running on Linux only (at least for now), but will implement your  suggestion for easier extension in the future.
The name is actually used only for creation of the PDF code on the server and not passed to the user or her OS.